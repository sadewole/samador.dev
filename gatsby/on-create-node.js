const { createFilePath } = require(`gatsby-source-filesystem`)

const onCreateNode = ({ node, actions, getNode }) => {
  const { createNodeField } = actions

  if (node.internal.type === `MarkdownRemark`) {
    if (typeof node.frontmatter.slug !== "undefined") {
      const dirname = getNode(node.parent).relativeDirectory
      createNodeField({
        node,
        name: "slug",
        value: `/${dirname}/${node.frontmatter.slug}`,
      })
    } else {
      const value = createFilePath({ node, getNode })
      createNodeField({
        node,
        name: "slug",
        value,
      })
    }
  }
}

module.exports = onCreateNode
