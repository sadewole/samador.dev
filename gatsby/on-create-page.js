const path = require(`path`)
const createBlogs = require("./pagination/create-blogs")

const createPages = async ({ graphql, actions, reporter }) => {
  const { createPage } = actions

  // Posts and pages from markdown
  const result = await graphql(`
    {
      allMarkdownRemark(filter: { frontmatter: { draft: { ne: true } } }) {
        nodes {
          frontmatter {
            slug
            layout
            title
          }
        }
      }
    }
  `)

  if (result.errors) {
    reporter.panicOnBuild(`There was an error loading this page`, result.errors)
    return
  }

  const { nodes } = result.data.allMarkdownRemark

  nodes.forEach(node => {
    if (node.frontmatter.layout === "page") {
      createPage({
        path: node.frontmatter.slug,
        component: path.resolve("./src/templates/page-template.js"),
        context: { slug: node.frontmatter.slug },
      })
    } else if (node.frontmatter.layout === "blog") {
      createPage({
        path: `/blogs/${node.frontmatter.slug}`,
        component: path.resolve("./src/templates/blog-template.js"),
        context: { slug: node.frontmatter.slug },
      })
    }
  })

  await createBlogs(graphql, actions)
}

module.exports = createPages
