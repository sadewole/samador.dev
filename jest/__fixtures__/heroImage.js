module.exports = {
  data: {
    mobileImage: {
      childImageSharp: {
        fixed: {
          base64: "test",
          height: 200,
          src: "photo.jpg",
          srcSet: "photo.jpg",
          width: 500,
        },
      },
    },
    desktopImage: {
      childImageSharp: {
        fixed: {
          base64: "test",
          height: 200,
          src: "photo1.jpg",
          srcSet: "photo1.jpg",
          width: 500,
        },
      },
    },
  },
}
