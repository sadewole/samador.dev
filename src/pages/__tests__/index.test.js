import React from "react"
import { render } from "@testing-library/react"
import renderer from "react-test-renderer"
import { StaticQuery, useStaticQuery } from "gatsby"

import Home from "../index"
import siteMetaData from "../../../jest/__fixtures__/site-metadata"
import heroImage from "../../../jest/__fixtures__/heroImage"

describe("Home page", () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }) => render(siteMetaData),
      useStaticQuery.mockReturnValue(siteMetaData)
    )
  })

  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      value: () => ({
        matches: false,
        addListener: () => {},
        removeListener: () => {},
      }),
    })
  })

  test("must be text", () => {
    const { getByTestId, getByText } = render(<Home {...heroImage} />)

    expect(getByText("Hi, I'm")).not.toBeNull()

    expect(getByTestId("author")).toHaveTextContent("Test name")
  })

  test("render correctly", () => {
    const tree = renderer.create(<Home {...heroImage} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
