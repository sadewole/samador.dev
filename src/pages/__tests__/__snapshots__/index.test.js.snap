// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page render correctly 1`] = `
<section
  className="relative py-10 bg-gray-700 dark:bg-gray-900 m-0 overflow-hidden h-screen"
>
  <div
    className="container h-full mx-auto flex items-center justify-center"
  >
    <div
      className="w-full px-14 z-3"
    >
      <h3
        className="leading-snug text-white font-light"
      >
        Hi, I'm
         
        <span
          aria-label="emoji"
          role="img"
        >
          ⛷️
        </span>
      </h3>
      <h1
        className="md:text-5xl text-3xl mb-2 leading-normal font-semibold text-white"
        data-testid="author"
      >
        Test name
      </h1>
      <p
        className="text-lg font-light leading-relaxed mt-0 mb-4 text-gray-300"
      >
        Test bio
      </p>
      <ul />
      <button
        aria-hidden="true"
        className={null}
        onClick={[Function]}
        style={
          Object {
            "MozAppearance": "none",
            "WebkitAppearance": "none",
            "appearance": "none",
            "backgroundColor": "transparent",
            "border": "none",
            "cursor": "pointer",
            "height": "37.599999999999994px",
            "overflow": "hidden",
            "padding": 0,
            "width": "80px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": "132px",
              "justifyContent": "center",
              "marginLeft": "-25.6px",
              "marginTop": "-47.599999999999994px",
              "width": "132px",
            }
          }
        >
          <div
            aria-label="animation"
            role="button"
            style={
              Object {
                "height": "100%",
                "margin": "0 auto",
                "outline": "none",
                "overflow": "hidden",
                "width": "100%",
              }
            }
            tabIndex={0}
            title=""
          />
        </div>
      </button>
      <nav
        style={
          Object {
            "marginTop": "-50px",
          }
        }
      >
        <ul>
          <li
            className="text-gray-300"
          >
            -
             
            <a
              className="hover:underline font-light leading-snug mr-2"
              href="/test/1/"
            >
              Test label 1
            </a>
          </li>
          <li
            className="text-gray-300"
          >
            -
             
            <a
              className="hover:underline font-light leading-snug mr-2"
              href="/test/2/"
            >
              Test label 2
            </a>
          </li>
          <li
            className="text-gray-300"
          >
            -
             
            <a
              className="hover:underline font-light leading-snug mr-2"
              href="/test/3/"
            >
              Test label 3
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      className="w-full md:w-4/12 px-4 mr-auto ml-auto mt-32 relative hidden md:block"
    >
      <div
        className="bg-gray-800 absolute -top-225-px -right-225-px rounded-full opacity-80 hidden md:block gatsby-image-wrapper"
        style={
          Object {
            "display": "inline-block",
            "height": 200,
            "overflow": "hidden",
            "position": "relative",
            "width": 500,
          }
        }
      >
        <picture>
          <source
            media="(min-width: 800px)"
            srcSet="test"
          />
          <source
            srcSet="test"
          />
          <img
            alt=""
            aria-hidden={true}
            src="test"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 1,
                "position": "absolute",
                "top": 0,
                "transitionDelay": "500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <picture>
          <source
            media="(min-width: 800px)"
            srcSet="photo1.jpg"
          />
          <source
            srcSet="photo.jpg"
          />
          <img
            alt=""
            height={200}
            loading="lazy"
            onLoad={[Function]}
            src="photo.jpg"
            srcSet="photo.jpg"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
            width={500}
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source media=\\"(min-width: 800px)\\" srcset=\\"photo1.jpg\\" /><source srcset=\\"photo.jpg\\" /><img loading=\\"lazy\\" width=\\"500\\" height=\\"200\\" srcset=\\"photo.jpg\\" src=\\"photo.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </div>
</section>
`;
